localstack/services/kms/provider.py

```diff
@@ -133,89 +163,238 @@ def __init__(self, message=None):
         super().__init__("ValidationError", message=message)
 
 
-class KmsProvider(KmsApi):
-    @staticmethod
-    def get_kms_backend(context: RequestContext) -> BaseBackend:
-        return kms_backends[context.account_id][context.region]
+class KmsProvider(KmsApi, ServiceLifecycleHook):
+    def _get_store(self, context: RequestContext) -> KmsStore:
+        return kms_stores[context.account_id][context.region]
+
+    def _get_key(self, context: RequestContext, key_id: str) -> KmsKey:
+        return self._get_store(context).get_key(key_id)
 
     @handler("CreateKey", expand=False)
     def create_key(
         self,
         context: RequestContext,
-        create_key_request: CreateKeyRequest = None,
+        request: CreateKeyRequest = None,
     ) -> CreateKeyResponse:
-        result = call_moto(context)
-
-        # generate keypair for signing, if this is a SIGN_VERIFY key
-        key_usage = create_key_request.get("KeyUsage", "ENCRYPT_DECRYPT")
-        key_spec = create_key_request.get("KeySpec", "SYMMETRIC_DEFAULT")
-        if not (key_usage == "ENCRYPT_DECRYPT" and key_spec == "SYMMETRIC_DEFAULT"):
-            create_key_request["KeyId"] = result["KeyMetadata"]["KeyId"]
-            _generate_data_key_pair(create_key_request, create_cipher=False)
-
-        return result
+        key = self._get_store(context).create_key(request, context.account_id, context.region)
+        return CreateKeyResponse(KeyMetadata=key.metadata)
+
+    @handler("ScheduleKeyDeletion", expand=False)
+    def schedule_key_deletion(
+        self, context: RequestContext, request: ScheduleKeyDeletionRequest
+    ) -> ScheduleKeyDeletionResponse:
+        pending_window = int(request.get("PendingWindowInDays", 30))
+        if pending_window < 7 or pending_window > 30:
+            raise ValidationException(
+                f"PendingWindowInDays should be between 7 and 30, but it is {pending_window}"
+            )
+        key = self._get_key(context, request.get("KeyId"))
+        key.schedule_key_deletion(pending_window)
+        attrs = ["DeletionDate", "KeyId", "KeyState"]
+        result = select_attributes(key.metadata, attrs)
+        result["PendingWindowInDays"] = pending_window
+        return ScheduleKeyDeletionResponse(**result)
+
+    @handler("CancelKeyDeletion", expand=False)
+    def cancel_key_deletion(
+        self, context: RequestContext, request: CancelKeyDeletionRequest
+    ) -> CancelKeyDeletionResponse:
+        key = self._get_key(context, request.get("KeyId"))
+        key.metadata["KeyState"] = "Disabled"
```

Comment:
Could use `KeyState.Disabled` here instead of "Disabled" to fulfil the typing requirements imposed by the generated ASF API

Reply:
Done.